contracts/test/behaviour/strategy.js

```diff
@@ -0,0 +1,401 @@
+const { expect } = require("chai");
+const { Wallet } = require("ethers");
+
+const { units, usdcUnits } = require("../helpers");
+const { impersonateAndFund } = require("../../utils/signers");
+const { parseUnits } = require("ethers/lib/utils");
+
+/**
+ *
+ * @param {*} context a function that returns a fixture with the additional properties:
+ * - strategy: the strategy to test
+ * - assets: array of tokens to test
+ * - valueAssets: optional array of assets that work for checkBalance and withdraw. defaults to assets
+ * - vault: Vault or OETHVault contract
+ * - harvester: Harvester or OETHHarvester contract
+ * @example
+    shouldBehaveLikeStrategy(() => ({
+      ...fixture,
+      strategy: fixture.fraxEthStrategy,
+      assets: [fixture.frxETH, fixture.weth],
+      valueAssets: [fixture.frxETH],
+      harvester: fixture.oethHarvester,
+      vault: fixture.oethVault,
+    }));
+ */
+const shouldBehaveLikeStrategy = (context) => {
+  describe("Strategy behaviour", () => {
+    it("Should have vault configured", async () => {
+      const { strategy, vault } = await context();
+      expect(await strategy.vaultAddress()).to.equal(vault.address);
+    });
+    it("Should be a supported asset", async () => {
+      const { assets, strategy } = await context();
+
+      for (const asset of assets) {
+        expect(await strategy.supportsAsset(asset.address)).to.be.true;
+      }
+    });
+    it("Should NOT be a supported asset", async () => {
+      const {
+        assets,
+        strategy,
+        usdt,
+        usdc,
+        dai,
+        weth,
+        reth,
+        stETH,
+        frxETH,
+        crv,
+        cvx,
+      } = await context();
+
+      const randomAssets = [
+        usdt,
+        usdc,
+        dai,
+        weth,
+        reth,
+        stETH,
+        frxETH,
+        cvx,
+        crv,
+      ];
+      for (const asset of randomAssets) {
+        if (assets.includes(asset)) {
+          continue; // Since `assets` already has a list of supported assets
+        }
+        expect(await strategy.supportsAsset(asset.address)).to.be.false;
```

Comment:
in the Curve strategies I've moved away from using `assetsMapped` in preference for more gas efficient immutables.

Reply:
Agree immutables are much better. Might be still worth reconsidering making them public instead of internal. Was the reasoning behind that maybe the build contract size?